File: internal/domain/repository.go
package domain

import "context"

type Repository interface {
	LockAndMarkProcessing(ctx context.Context, limit int32) ([]VideoJob, error)
	MarkDone(ctx context.Context, id int64) error
	MarkRetry(ctx context.Context, id int64, delay int) error
	MarkFailed(ctx context.Context, id int64) error
	Exists(ctx context.Context, id int64) (bool, error)
}

File: domain/repository.ts
import type { VideoJob } from "./job.js";

export interface Repository {
    lockAndMarkProcessing(limit: number): Promise<VideoJob[]>;
    markDone(id: bigint): Promise<void>;
    markRetry(id: bigint, delay: number): Promise<void>;
    markFailed(id: bigint): Promise<void>;
    exists(id: bigint): Promise<boolean>;
}